packages/driver/src/cy/aliases.ts

```diff
@@ -16,16 +20,19 @@ const aliasDisplayName = (name) => {
 // eslint-disable-next-line @cypress/dev/arrow-body-multiline-braces
 export const create = (cy: $Cy) => ({
   addAlias (ctx, aliasObj) {
-    const { alias, subject } = aliasObj
+    const { alias } = aliasObj
 
     const aliases = cy.state('aliases') || {}
 
     aliases[alias] = aliasObj
     cy.state('aliases', aliases)
 
-    const remoteSubject = cy.getRemotejQueryInstance(subject)
-
-    ctx[alias] = remoteSubject ?? subject
+    Object.defineProperty(ctx, alias, {
+      configurable: true,
+      get () {
+        return $utils.getSubjectFromChain(aliasObj.subjectChain, cy)
+      },
+    })
```

Comment:
Won't this break if a user is enumerating aliases on `this`? Or any use case where the user does `.as('foo')` and then later does `this.foo = ...`?

Like this: ![image](https://user-images.githubusercontent.com/1151760/189746268-eb33e4ac-5b43-42ce-a516-57d2c304bf5a.png)

Reply:
Yeah; I actually worked around this in one of our very own tests, where we have `.as('win'); cy.then(() => { this.win = ...override the alias we just made... })`.

This particular change isn't necessary - I could leave it as 'set on context once when command executes', but it seemed like an improvement. Setting aliases as static properties on the context suffers from the same "can be out of date" issue as any other cypress command.

Reply:
Would something like this work?

```suggestion
    let hasSetValue = false
    let ctxAliasValue: any = undefined
    Object.defineProperty(ctx, alias, {
      configurable: true,
      enumerable: true,
      get () {
        return hasSetValue ? ctxAliasValue : $utils.getSubjectFromChain(aliasObj.subjectChain, cy)
      },
      set (value) {
        hasSetValue = true
        ctxAliasValue = value
      }
    })
```

I also agree the spirit of this change is an improvement in the direction of detached DOM, but I feel like this should be removed for now, given the high potential to cause breakage. And we can release this improvement in a breaking version.

Reply:
I'll remove this from this PR and add it to the next one I'm working on (which will target 11.x rather than develop).

Good thought about accounting for overwrites, will include something to that effect.